metricbeat/module/redis/info/info.go

```diff
@@ -43,45 +45,57 @@ func (m *MetricSeter) Setup(ms *helper.MetricSet) error {
 	}
 
 	for _, host := range ms.Config.Hosts {
+		redisPool := createPool(host, config.Password, config.Network, config.MaxConn, ms)
+		m.redisPools[host] = redisPool
+	}
+
+	return nil
+}
 
-		// Set up redis pool
-		redisPool := rd.NewPool(func() (rd.Conn, error) {
-			// Sets timeout to exit request if is longer then Timeout
-			c, err := rd.Dial(config.Network, host, rd.DialReadTimeout(ms.Module.Timeout))
+func createPool(host, password, network string, maxConn int, ms *helper.MetricSet) *rd.Pool {
 
+	return &rd.Pool{
+		MaxIdle:     maxConn,
+		IdleTimeout: ms.Module.Timeout,
+		Dial: func() (rd.Conn, error) {
+			c, err := rd.Dial(network, host)
 			if err != nil {
-				logp.Err("Failed to create Redis connection pool: %v", err)
 				return nil, err
 			}
-
+			if password != "" {
+				if _, err := c.Do("AUTH", password); err != nil {
+					c.Close()
+					return nil, err
+				}
+			}
 			return c, err
-		}, config.MaxConn)
-
-		// TODO: add AUTH
-		m.redisPools[host] = redisPool
+		},
 	}
-
-	return nil
 }
 
 func (m *MetricSeter) Fetch(ms *helper.MetricSet, host string) (events common.MapStr, err error) {
 
 	// Fetch default INFO
-	info := m.fetchRedisStats(host, "default")
-	event := eventMapping(info)
-	return event, nil
+	info, err := m.fetchRedisStats(host, "default")
+
+	if err != nil {
+		return nil, err
+	}
+
+	return eventMapping(info), nil
 }
 
 // fetchRedisStats returns a map of requested stats
-func (m *MetricSeter) fetchRedisStats(host string, stat string) map[string]string {
+func (m *MetricSeter) fetchRedisStats(host string, stat string) (map[string]string, error) {
 	c := m.redisPools[host].Get()
 	out, err := rd.String(c.Do("INFO", stat))
 	c.Close()
```

Comment:
Agreed. I think it wasn't there before, since `info.Fetch()` used to iterate over the hosts internally...